doc/GUIDE.md

```diff
@@ -0,0 +1,822 @@
+and search mechanism can be used as well. When the `--use-system-load-library` option is
+specified to jextract, the generated bindings will try to load libraries specified using
+`--library` through [`System::loadLibrary`](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/System.html#loadLibrary(java.lang.String)).
+The library search path for `System::loadLibrary` is specified through the [`java.library.path`](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/System.html#java.library.path)
+system property instead of the OS-specific environment variable. Though, please note
+that if the loaded library has any dependencies, those dependencies will again be loaded
+through the OS-specific library loading mechanism (this is outside of the JVM's control).
+
+In both cases, the library is unloaded when the class loader that loads the binding
+classes is garbage collected.
+
+## Using The Code Generated By Jextract
+
+In the following section we'll go over examples of declarations that might be found in a C
+header file, show the code that jextract generates for these declarations, and show
+some examples of how to use the generated Java code.
+
+Most of the code that jextract generates will be available through a single class. By default
+the name of that class is derived from the name of the main header file that is passed to
+jextract. For example, if the header file is named `mylib.h`, then the derived class name
+will be `mylib_h`. The class name can be set explicitly using the `--header-class-name`
+option as well.
+
+Besides the main header class that is generated, jextract also generates several
+separate files for certain declarations in the C header file. Namely, structs, unions,
+function pointer types, and typedefs of struct or union types result in additional files
+being generated. We will look at those in more detail in this section.
+
+Most of the methods that jextract generates are `static`, and are designed to be imported
+using `import static`. Typically, to access the code that jextract generates for a header
+file called `mylib.h`, only the following two wildcard imports are needed:
+
+```java
+import static org.mypackage.mylib_h.*;
+import org.mypackage.*;
+```
+
+Where `org.mypackage` is the package into which jextract put the generates source files
+(using `--target-package`/`-t`).
+
+The former import statement will import all the static functions and fields from the class
+that jextract generates for the main header file of the library. This includes methods to
+access functions, global variables, macros, enums, primitive typedefs, and layouts for
+builtin C types.
+
+The latter import statement imports all the other classes the jextract generates, which
+includes: classes representing structs or unions, function types, and struct or union
+typedefs.
+
+### Builtin Type Layouts
+
+For every jextract run, regardless of the contents of the library header files, jextract
```

Comment:
I'm not sure it's clear where these layouts definitions are introduced. I think the text should say that these are appended to the main header class.

Reply:
Perhaps you can add a comment:
```
// mylib_h.java
```
as you do in later examples?